File: src/main/java/dev/liquid/osamaclient/event/PacketEvent.java
package dev.liquid.osamaclient.event;

import net.minecraft.network.Packet;
import net.minecraftforge.fml.common.eventhandler.Event;


public class PacketEvent {
  public static class Received extends Event {
    public Packet<?> packet;

    public Received(Packet<?> packet) {
      this.packet = packet;
    }
  }

  public static class Sent extends Event {
    public Packet<?> packet;

    public Sent(Packet<?> packet) {
      this.packet = packet;
    }
  }
}

File: src/main/kotlin/dev/liquid/osamaclient/event/PacketEvent.kt
package dev.liquid.osamaclient.event

import net.minecraft.network.Packet;
import net.minecraftforge.fml.common.eventhandler.Event;

class PacketEvent {
  class Received(var packet: Packet<*>) : Event()
  class Sent(var packet: Packet<*>) : Event()
}
